QGC Parameters

afwilkin opened the pull request:
When you want to download a parameter file... it can not be directly uploaded into QGC. This givces you the option to download the QGC compatible one. This PR was based off of @farhangnaderi PR here: #264

afwilkin commented:
@bkueng let me know if there is any changes you would like me to make

farhangnaderi commented:
Thanks Andrew! I appreciate it!

bkueng commented on app/plot_app/templates/header.html:
```diff
@@ -56,7 +56,9 @@ <h1>
 			<div class="dropdown-menu">
 				<a class="dropdown-item" href="download?log={{ log_id }}">Log File</a>
 				<a class="dropdown-item" href="download?log={{ log_id }}&type=1" target="_blank">Parameters</a>
```
I think we can remove this option (but keep the implementation), and just offer the QGC versions.

bkueng commented on app/tornado_handlers/download.py:
```diff
@@ -155,6 +154,75 @@ def kml_colors(flight_mode):
+
+            delimiter = '	'
+            for param_key in param_keys:
+                self.write("1")
```
Can you add a comment that this is the sysid (and compid below)?

bkueng commented on app/tornado_handlers/download.py:
```diff
@@ -155,6 +154,75 @@ def kml_colors(flight_mode):
+                    except:
+                        pass
+
             else:
```
This `else` case was part of the `elif download_type == '3': # download the non-default parameters` case above and needs to be copied up to that case again, and here it needs to be adjusted for the QGC format.

bkueng reviewed (commented):
There's also a couple of CI problems. We should start to split the long method into individual methods by download type.

afwilkin commented:
Apologies Beat, I did not see you had reviewed this. I will fix these

afwilkin commented:
@bkueng I addressed your comments. I know you mentioned there are some CI problems that we will incur. Mainly the long method. what is the method length that will pass the CICD test so I know how to modify the code?

afwilkin commented:
@bkueng made changes. 

Also added something to the bottom of the readme to let others know to run `./run_pylint` before making PR to help with CICD.

bkueng reviewed (approved):
Thanks
